packages/manager/cypress/e2e/core/longview/longview.spec.ts

```
@@ -100,85 +73,133 @@ const waitForLongviewData = (
-
-      // Sometimes Cloud Manager UI does not updated automatically upon receiving
-      // Longivew status data. Performing a page reload mitigates this issue.
-      // TODO Remove call to `cy.reload()`.
-      cy.reload();
-      cy.get(`[data-qa-longview-client="${client.id}"]`)
-        .should('be.visible')
-        .within(() => {
-          cy.findByText('Waiting for data...').should('not.exist');
-          cy.findByText('CPU').should('be.visible');
-          cy.findByText('RAM').should('be.visible');
-          cy.findByText('Swap').should('be.visible');
-          cy.findByText('Load').should('be.visible');
-          cy.findByText('Network').should('be.visible');
-          cy.findByText('Storage').should('be.visible');
-        });
+    mockGetLongviewClients([client]).as('getLongviewClients');
+    mockFetchLongviewStatus(client, 'lastUpdated', longviewLastUpdatedWaiting);
+    mockFetchLongviewStatus(client, 'getValues', longviewGetValuesWaiting);
+    mockFetchLongviewStatus(
+      client,
+      'getLatestValue',
+      longviewGetLatestValueWaiting
+    ).as('fetchLongview');
+
+    const installCommand = getInstallCommand(client.install_code);
+
+    cy.visitWithLogin('/longview');
+    cy.wait('@getLongviewClients');
+
+    // Confirm that Longview landing page lists a client that is still waiting for data...
+    cy.get(`[data-qa-longview-client="${client.id}"]`)
+      .should('be.visible')
+      .within(() => {
+        cy.findByText(client.label).should('be.visible');
+        cy.findByText(client.api_key).should('be.visible');
+        cy.contains(installCommand).should('be.visible');
+        cy.findByText('Waiting for data...');
+      });
+
+    // Update mocks after initial Longview fetch to simulate client installation and data retrieval.
+    // The next time Cloud makes a request to the fetch endpoint, data will start being returned.
+    cy.wait(['@fetchLongview', '@fetchLongview', '@fetchLongview']).then(() => {
```

Can we reduce this to 2 or even 1 fetch so the test doesn't take as long to complete? Or is this necessary? Same for line 237.

---

3 fetches here was intended to avoid some intermittent issue that was observed during local testing. Let me try to remove some and test again in the jenkins to see if it is stable enough.

---

I'll follow up later this afternoon with an explanation why this is necessary, but I suspect that removing any of them may result in inconsistencies with the mock data that the Longview endpoint responds with (which may not necessarily manifest as a failure)

---

Sorry for the slow follow up!

This is necessary because the Longview landing page, for each client, fires 3 requests to the Longview fetch endpoint when the page loads (and then repeats this process at some interval, kind of like how we poll for events). It's worth pointing out that all 3 of these requests share the same endpoint, but they have a different outgoing payload to specify e.g. what action is being performed or what data is being requested.

Because these all hit the same endpoint, we can't use Cypress to assign an alias to each request -- if we did, they would all satisfy one another. So instead we have to assign one alias to one of our intercepts, and wait for the alias to resolve 3 times so that we can ensure that our _next_ intercepts get set up correctly (e.g. the right data being mocked for the right request). If we don't wait for all 3, depending on how quickly Cypress is executing, it's possible for the next mock data to get set up before the 2nd or 3rd request resolves, causing the landing page to behave as if it's receiving data when the test expects it to be in its "waiting for data..." state.

This weird and careful sequencing of events is only really necessary because all these requests hit the same endpoint, but it's worth noting that there isn't really a serious performance penalty to doing this -- Cloud fires these 3 requests all at once, and they resolve quickly, so waiting for all 3 isn't adding any perceivable time to the test duration, it just ensures that Cypress doesn't proceed a fraction of a second too quickly.

---

Ahh this makes a lot of sense. It might be worth including a comment including a brief explanation (or even a link to this comment, lol). Thanks for following up with the detailed explanation!